doxygen/parallelism_support/opencl_support.md

```diff
@@ -94,20 +94,128 @@ Once you have done the above step, `runTests.py` should execute with the GPU ena
 
 ## Using the OpenCL backend
 
-We currently have support for the following methods
+The OpenCL backend can be used for reverse mode AD as well as primitive functions on containers of basic C++ scalar types. Below is the list of functions and distributions that are currently supported.
+
+### Reverse mode
+
+An example of using OpenCL supported Stan Math functions for reverse mode AD is shown below: 
+
+```cpp
+using stan::math;
+
+Eigen::Matrix<double, -1, -1> A(N,M);
+Eigen::Matrix<double, -1, -1> B(M,N);
+// ... fill matrices A and B
+var_value<matrix_cl<double>> A_cl = to_matrix_cl(A);
+var_value<matrix_cl<double>> B_cl = to_matrix_cl(B);
+var_value<matrix_cl<double>> C_cl = A_cl * B_cl;
+var_value<Eigen::Matrix<double, -1, -1>> C = from_matrix_cl(C_cl);
+C(0,0).grad();
```

Comment:
Here we would probably want to set C's values to 1 and then run `grad()` right? @bbbales2 what was the reason we don't have `C.grad()` that fills the adjoints with 1 by default for var matrix types?

Reply:
Added the sum() to avoid this.